ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -275,16 +275,21 @@ void writeString(int FD, string S) {
   const char *Buf = S.c_str();
 
   size_t Remain = strlen(Buf);
+
+  // Otherwise we can't reliably know how much was written!
+  if (Remain > SSIZE_MAX)
+    errx(EXIT_FAILURE, "Remain > SSIZE_MAX");
+
   while (Remain) {
-    assert(Remain >= 0);
-    int Wrote = write(FD, Buf, Remain);
+    ssize_t Wrote = write(FD, Buf, Remain);
     if (Wrote == -1) {
       if (errno == EINTR)
         continue;
       else
         err(EXIT_FAILURE, "write");
     }
 
+    assert(static_cast<size_t>(Wrote) <= Remain);
```

Comment:
Note that if a silly implementation of `write` returns something < -1 then the `static_cast` will (I assume!) throw the signedness away. You could do `debug_assert(Wrote >= 0 && static_cast<size_t>(Wrote) <= Remain)`?

[Dunno if C++ has `debug_assert` but, if so, I'd be fine with using that. I don't think we need to `assert` this in release mode all the time.]

Reply:
if `write` returned `-1` we don't get to this portion of code.

In C++ `assert()` is a `debug_assert!()`. I opted to use a hard error with `err()` as we probably want to know if this happens in release mode.

Reply:
Oh wait. `< -1`, let me think.

Reply:
Note: I'm suggesting a stronger check that `write` doesn't return a value lower than `-1`. If that happened, the `static_cast<size_t>` would just remove that information, which would be confusing.

Reply:
I'll attempt a fix, but if a platform doesn't do POSIX, then you probably have other issues :P

Reply:
I think `assert(Wrote >= 0 && static_cast<size_t>(Wrote) <= Remain)` is easy to understand and catches the < -1 horror?

Reply:
Fixed in fc913f5..4533234

Reply:
Genius code :p